common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -739,6 +739,30 @@ bool mon_wlan_hal_dwpal::sta_beacon_11k_request(const SBeaconRequest11k &req, in
         cmd += " ssid=" + req_ssid;
     }
 
+    // use ap_ch_report in the request.
+    if (req.use_optional_ap_ch_report) {
+        std::stringstream ch_report_ss;
+        uint8_t current_channel = 0;
+        for (; current_channel < req.use_optional_ap_ch_report - 1; ++current_channel) {
```

Comment:
After discussion in zoom: let's just keep the structure as it is and use it as a length, and only change the comment in the yaml file so it says length instead of boolean.